### ISSUE TITLE
`eth_getLogs` not working as expected

### ISSUE THREAD

**Issue by gluneau** [AcalaNetwork/bodhi.js#158]
ganache response to `eth_getLogs`:
```
curl http://localhost:8545 \
    -X POST \
    -H "Content-Type: application/json" \
    -d '{"jsonrpc":"2.0","method":"eth_getLogs","params": [{ "fromBlock": "0x4", "toBlock": "0x5", "topics": []}],"id":1}' 
{"id":1,"jsonrpc":"2.0","result":[{"logIndex":"0x0","transactionIndex":"0x0","transactionHash":"0xa3efd4fc42f9c7184f23ec4d7b62b626521e3dda3768a8e61bc87e0ec953b097","blockHash":"0x62850366828e24eeb0c0df6b7a010a24b926278951d9195f386dbff2a9c92c7b","blockNumber":"0x4","address":"0x217b896620aff6518b9862160606695607a63442","data":"0x","topics":["0x8be0079c531659141344cd1fd0a4f28419497f9722a3daafe3b4186f6b6457e0","0x0000000000000000000000000000000000000000000000000000000000000000","0x00000000000000000000000075e480db528101a381ce68544611c169ad7eb342"],"type":"mined","removed":false}]}%  
```

bodhi.js response
```
curl http://localhost:8545 \
    -X POST \
    -H "Content-Type: application/json" \
    -d '{"jsonrpc":"2.0","method":"eth_getLogs","params": [{ "fromBlock": "0x4", "toBlock": "0x5", "topics": []}],"id":1}'
{"id":1,"jsonrpc":"2.0","error":{"code":-32602,"message":"blocktag should be number | 'latest' | 'earliest' (argument=\"blockTag\", value=\"0x4\", code=INVALID_ARGUMENT, version=bodhi.js/providers/2.1.4)"}}% 
```

https://github.com/AcalaNetwork/bodhi.js/issues/95
https://github.com/AcalaNetwork/bodhi.js/pull/104

**Comment by gluneau** [comment]
**TL;DR, By comparing these two data, you can find normal eth rpc use filed type hex, such as field: "blockNumber","logIndex","transactionIndex".**

> We find some rpc query on acala do not work, it due to the resp strcut from acala.
> For most normal evm rpc, it use hex data type. 
> For example, we call evm chain such as ethereum,bsc,arb, eth by rpc "eth_getLogs".

The resp is always like this:
```
{
 "jsonrpc": "2.0",
 "id": 1,
 "result": [{
  "address": "0x358234b325ef9ea8115291a8b81b7d33a2fa762d",
  "blockHash": "0xe67aa40addf64a7a0850922dc160b72ba67c4071570922960d77941d8cb900a0",
  "blockNumber": "0x5fa016",
  "data": "0x7ebf0e016765a1fa31a3d45620cc673ba04c48baf4b23adeb09982c823a977f100000000000000000000000051d36e18e3d32d121a3cfe2f3e5771a6fd53274e00000000000000000000000051d36e18e3d32d121a3cfe2f3e5771a6fd53274e000000000000000000000000f4b2cbc3ba04c478f0dc824f4806ac39982dce730000000000000000000000000000000000000000000000000000000011d991500000000000000000000000000000000000000000000000000000000000000061196eb8416e00bc51ca8dc80e346e4581652b7a9874e6238015c6d72dc2aee8aa",
  "logIndex": "0xa",
  "removed": false,
  "topics": ["0x79fa08de5149d912dce8e5e8da7a7c17ccdf23dd5d3bfe196802e6eb86347c7c"],
  "transactionHash": "0x4e5924bcb72cdcde4887f130f7b312ae9a2c1505caf82b9a82276f461877e98e",
  "transactionIndex": "0x19"
 }]
}
```

While we call acala rpc with "eth_getLogs", it always returns resp below.
```
[{
 "blockNumber": 422586,
 "blockHash": "0xa1a07ccf1bb31e8da1e1d62cb5aecd3012f8596826ce750f976d7f1fdfb542a5",
 "transactionIndex": 0,
 "removed": false,
 "address": "0xe6F4a83eE9f946B86a2ef008dCD872f4a942DB24",
 "data": "0x426ab38338be5d42c2cafd1075db80e71965e43f87c11536d8cf0a0dae40d54300000000000000000000000029b563951ed0eb9ae5c49692266e1fbc81445cfe00000000000000000000000029b563951ed0eb9ae5c49692266e1fbc81445cfe00000000000000000000000064ff10dced55d2efeb47f132dd09d37616bfbd18000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000002532c0a3b36bf2a0bbe9a9e96ceeb0ef55ed415dfaafae68823b57bcbffab4e0cca",
 "topics": ["0x79fa08de5149d912dce8e5e8da7a7c17ccdf23dd5d3bfe196802e6eb86347c7c"],
 "transactionHash": "0x20220cf9d4bf9a666fc7507b47ae85339a81a899c958a83af644453243c86603",
 "logIndex": 1
}]
```
